treatment-service/models/storage.go

```diff
@@ -5,22 +5,24 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
 	"log"
 	"net/http"
+	"os"
 	"strings"
 	"sync"
 	"time"
 
-	"github.com/golang-collections/collections/set"
-	"golang.org/x/oauth2/google"
-
 	managementClient "github.com/caraml-dev/xp/clients/management"
 	"github.com/caraml-dev/xp/common/api/schema"
 	"github.com/caraml-dev/xp/common/pubsub"
 	_segmenters "github.com/caraml-dev/xp/common/segmenters"
+	"github.com/golang-collections/collections/set"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 )
 
+const ExpGoogleApplicationCredentials = "EXP_GOOGLE_APPLICATION_CREDENTIALS"
```

Comment:
This is reasonable, but it's a bit strange to let this common code be driven by Turing's choice of env var for `EXP_GOOGLE...`. I think it will be better to make it a config (perhaps in the [DeploymentConfig](https://github.com/caraml-dev/xp/blob/main/treatment-service/config/config.go#L72) struct, as `GoogleApplicationCredentialsEnvVar` - this config is applicable to PubSub / Management API calls now, later on to BQ logging of the treatment generation as well), much like other config values that can vary across deployments. Its default value can be unset. This will still match the implementation you have in `treatment-service/services/subscription.go`.

Wdyt?

Reply:
That makes sense; I've just refactored all the implementations in the relevant services to read off the filepath from the `GoogleApplicationCredentialsEnvVar` config stored in the `DeploymentConfig`.